Translate Hooks FAQ

dummyeuy opened the pull request:
Translating content/docs/hooks-faq.md by dummyeuy with minor revisions

Referenced from this https://github.com/reactjs/id.reactjs.org/issues/1#issuecomment-538836217

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
+Hooks didesain dengan *static typing* pada dasarnya. Karena hal-hal tersebut adalah fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi *Flow* dan *TypeScript React* sudah termasuk *support* untuk React *Hooks* di dalamnya.
 
-Importantly, custom Hooks give you the power to constrain React API if you'd like to type them more strictly in some way. React gives you the primitives, but you can combine them in different ways than what we provide out of the box.
+Penting untuk diingat, *custom Hooks* memberikan Anda kewenangan untuk memaksa *React API* jika Anda ingin menuliskannya dengan lebih ketat dalam beberapa hal. React memberikan Anda *primitives*, tapi Anda bisa menggabungkannya dengan cara-cara lain yang unik selain dari yang kami sediakan untuk Anda.
```
```suggestion
Penting untuk diingat, *custom Hooks* memberikan Anda kewenangan untuk memaksa API React jika Anda ingin menuliskannya dengan lebih ketat dalam beberapa hal. React memberikan Anda *primitives*, tapi Anda bisa menggabungkannya dengan cara-cara lain yang unik selain dari yang kami sediakan untuk Anda.
```

dummyeuy replied:
>Penting untuk diingat, *custom Hooks* memberikan Anda kewenangan untuk memaksa *API* React jika Anda ingin menuliskannya dengan lebih ketat dalam beberapa hal. React memberikan Anda *primitives*, tapi Anda bisa menggabungkannya dengan cara-cara lain yang unik selain dari yang kami sediakan untuk Anda.

suggested revision, applied

resir014 commented on content/docs/hooks-faq.md:
````diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
   );
 }
 ```
+Kami akan mengujinya dengan React *DOM*. Pastikan bahwa perilakunya dengan yang terjadi di peramban, kami akan mengemas proses *rendering* kode dan memperbaruinya ke dalam [`ReactTestUtils.act()`](/docs/test-utils.html#act) beberapa pemanggilan:
````
```suggestion
Kami akan mengujinya dengan React DOM. Pastikan bahwa perilakunya dengan yang terjadi di peramban, kami akan mengemas proses *rendering* kode dan memperbaruinya ke dalam [`ReactTestUtils.act()`](/docs/test-utils.html#act) beberapa pemanggilan:
```

Abbreviations in foreign languages need not be italicised.
Reference: https://twitter.com/ivanlanin/status/898520073224167424

dummyeuy replied:
>Kami akan mengujinya dengan React DOM. Pastikan bahwa perilakunya dengan yang terjadi di peramban, kami akan mengemas proses *rendering* kode dan memperbaruinya ke dalam [`ReactTestUtils.act()`](/docs/test-utils.html#act) beberapa pemanggilan:

suggested revision, applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -164,71 +163,71 @@ afterEach(() => {
 
-* Calls to Hooks are either inside a `PascalCase` function (assumed to be a component) or another `useSomething` function (assumed to be a custom Hook).
-* Hooks are called in the same order on every render.
+* Panggilan untuk *Hooks* bisa terdapat di dalam sebuah fungsi *`PascalCase`* (diasumsikan sebagai sebuah komponen) atau fungsi *`useSomething`* lainnya (diasumsikan sebagai sebuah *custom Hook*).
```
```suggestion
* Panggilan untuk *Hooks* bisa terdapat di dalam sebuah fungsi *`PascalCase`* (diasumsikan sebagai sebuah komponen) atau fungsi `useSomething` lainnya (diasumsikan sebagai sebuah *custom Hook*).
```

Method names inside code blocks don't need to be italicised.

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -164,71 +163,71 @@ afterEach(() => {
+### Apakah ada hal seperti *instance variable*? {#is-there-something-like-instance-variables}
 
-Yes! The [`useRef()`](/docs/hooks-reference.html#useref) Hook isn't just for DOM refs. The "ref" object is a generic container whose `current` property is mutable and can hold any value, similar to an instance property on a class.
+Ya! [`useRef()`](/docs/hooks-reference.html#useref) Hook tidak hanya untuk *DOM ref*. Objek *"ref"* adalah sebuah *container* umum yang mana memiliki properti *`current`* yang berubah-ubah serta dapat memegang nilai apapun, serupa dengan *instance property* pada sebuah kelas.
```
```suggestion
Ya! [`useRef()`](/docs/hooks-reference.html#useref) Hook tidak hanya untuk *ref* DOM. Objek *"ref"* adalah sebuah *container* umum yang mana memiliki properti `current` yang berubah-ubah serta dapat memegang nilai apapun, serupa dengan *instance property* pada sebuah kelas.
```

dummyeuy replied:
>Ya! [`useRef()`](/docs/hooks-reference.html#useref) Hook tidak hanya untuk *ref* DOM. Objek *"ref"* adalah sebuah *container* umum yang mana memiliki properti `current` yang berubah-ubah serta dapat memegang nilai apapun, serupa dengan *instance property* pada sebuah kelas.

suggested revision, applied

dummyeuy commented on content/docs/hooks-faq.md:
```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
+* React *Native*
+* React *DOM Server*
+* React *Test Renderer*
+* React *Shallow Renderer*
```
Understood.
Revision's applied

dummyeuy commented on content/docs/hooks-faq.md:
```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
-  * [What is the prior art for Hooks?](#what-is-the-prior-art-for-hooks)
+* **[Strategi Adopsi](#adoption-strategy)**
+  * [Versi React mana yang termasuk *Hooks* di dalamnya?](#which-versions-of-react-include-hooks)
+  * [Apakah perlu menulis ulang semua kelas komponen saya?](#do-i-need-to-rewrite-all-my-class-components)
```
Understood and (suggested) revision's applied

dummyeuy commented on content/docs/hooks-faq.md:
```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
+### Apakah perlu menulis ulang semua kelas komponen saya?{#do-i-need-to-rewrite-all-my-class-components}
 
-No. There are [no plans](/docs/hooks-intro.html#gradual-adoption-strategy) to remove classes from React -- we all need to keep shipping products and can't afford rewrites. We recommend trying Hooks in new code.
+Tidak. [Tidak ada rencana](/docs/hooks-intro.html#gradual-adoption-strategy) untuk menghilangkan kelas dari React -- kami perlu terus mengirimkan produk dan tidak akan melakukan penulisan ulang. Kami rekomendasikan untuk mencoba *Hooks* dalam kode baru.
```
> Tidak. [Tidak ada rencana](/docs/hooks-intro.html#gradual-adoption-strategy) untuk menghilangkan kelas dari React -- kita semua perlu mengirimkan produk baru dan tidak memiliki waktu untuk penulisan ulang. Kami rekomendasikan untuk mencoba *Hooks* dalam kode baru.

Understood and (suggested) revision's applied

dummyeuy commented on content/docs/hooks-faq.md:
```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
+* **[Strategi Adopsi](#adoption-strategy)**
+  * [Versi React mana yang termasuk *Hooks* di dalamnya?](#which-versions-of-react-include-hooks)
+  * [Apakah perlu menulis ulang semua kelas komponen saya?](#do-i-need-to-rewrite-all-my-class-components)
+  * [Apa yang dapat saya lakukan dengn *Hooks* namun tidak bisa saya lakukan dengan kelas-kelas?](#what-can-i-do-with-hooks-that-i-couldnt-with-classes)
```
> [Apa yang dapat saya lakukan dengn *Hooks* namun tidak bisa saya lakukan dengan kelas?]

Suggested revision's applied

dummyeuy commented on content/docs/hooks-faq.md:
```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
+Tidak. [Tidak ada rencana](/docs/hooks-intro.html#gradual-adoption-strategy) untuk menghilangkan kelas dari React -- kami perlu terus mengirimkan produk dan tidak akan melakukan penulisan ulang. Kami rekomendasikan untuk mencoba *Hooks* dalam kode baru.
 
-### What can I do with Hooks that I couldn't with classes? {#what-can-i-do-with-hooks-that-i-couldnt-with-classes}
+### Apa yang dapat saya lakukan dengn *Hooks* namun tidak bisa saya lakukan dengan kelas-kelas? {#what-can-i-do-with-hooks-that-i-couldnt-with-classes}
```
> ### Apa yang dapat saya lakukan dengn *Hooks* namun tidak bisa saya lakukan dengan kelas? 

suggested revision, applied

dummyeuy commented on content/docs/hooks-faq.md:
```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
+### Apa yang dapat saya lakukan dengn *Hooks* namun tidak bisa saya lakukan dengan kelas-kelas? {#what-can-i-do-with-hooks-that-i-couldnt-with-classes}
 
-Hooks offer a powerful and expressive new way to reuse functionality between components. ["Building Your Own Hooks"](/docs/hooks-custom.html) provides a glimpse of what's possible. [This article](https://medium.com/@dan_abramov/making-sense-of-react-hooks-fdbde8803889) by a React core team member dives deeper into the new capabilities unlocked by Hooks.
+Hooks menawarkan sebuah cara baru yang kuat dan ekpresif untuk menggunakan kembali fungsionalitas antara komponen. ["Buat Hooks-mu Sendiri"](/docs/hooks-custom.html) menyajikan sekialas apa saja yang mungkin dilakukan. [Artikel ini](https://medium.com/@dan_abramov/making-sense-of-react-hooks-fdbde8803889), oleh seorang anggota inti tim React, menyelam lebih dalam lagi ke kemampuan-kemampuan baru  ...
```
> Hooks menawarkan sebuah cara baru yang kuat dan ekpresif untuk menggunakan kembali fungsionalitas antara komponen. ["Buat Hooks-mu Sendiri"](/docs/hooks-custom.html) menyajikan sekilas apa saja yang mungkin dilakukan. [Artikel ini](https://medium.com/@dan_abramov/making-sense-of-react-hooks-fdbde8803889), oleh seorang anggota inti tim React, menyelam lebih dalam lagi kemampuan-kemampuan baru yang dibuka oleh Hooks.

suggested revision, applied

dummyeuy commented on content/docs/hooks-faq.md:
```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
+### Apakah *Hooks* menggantikan *render props* dan *higher-order components*? {#do-hooks-replace-render-props-and-higher-order-components}
 
-Often, render props and higher-order components render only a single child. We think Hooks are a simpler way to serve this use case. There is still a place for both patterns (for example, a virtual scroller component might have a `renderItem` prop, or a visual container component might have its own DOM structure). But in most cases, Hooks will be sufficient and can help reduce nesting in your tre ...
+Seringkali, *render props* dan *higher-order components* hanya (me)*render* sebuah turunan. Kami pikir *Hooks* adalah cara yang lebih sederhana untuk menyajikan kasus penggunaan ini. Masih ada tempat untuk kedua pola (contohnya, sebuah *virtual scroller component* bisa saja memiliki sebuah *`renderItem` prop*, atau sebuah *visual container component* bisa saja memiliki struktur *DOM*-nya sendiri) ...
```
>Seringkali, *render props* dan *higher-order components* hanya (me)*render* sebuah turunan. Kami pikir *Hooks* adalah cara yang lebih sederhana untuk menyajikan kasus penggunaan ini. Masih ada tempat untuk kedua pola (contohnya, sebuah *virtual scroller component* bisa saja memiliki sebuah *prop* `renderItem`, atau sebuah *visual container component* bisa saja memiliki struktur *DOM*-nya sendiri). Namun dalam banyak kasus, *Hooks* sudahlah cukup dan akan membantu mengurangi *nesting* dalam tatanan Anda.

suggested revision, applied

dummyeuy commented on content/docs/hooks-faq.md:
```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
+### Apakah *Hooks* dapat berjalan dengan *static typing*?{#do-hooks-work-with-static-typing}
 
-Hooks were designed with static typing in mind. Because they're functions, they are easier to type correctly than patterns like higher-order components. The latest Flow and TypeScript React definitions include support for React Hooks.
+Hooks didesain dengan *static typing* pada dasarnya. Karena hal-hal tersebut adalah fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi *Flow* dan *TypeScript React* sudah termasuk *support* untuk React *Hooks* di dalamnya.
```
>*Hooks* didesain dengan *static typing* pada dasarnya. Karena hal-hal tersebut adalah fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi *Flow* dan *TypeScript React* untuk React sudah mendukung untuk React Hooks di dalamnya.

suggested revision, applied

dummyeuy commented:
Dear @resir014 my latest commit as you can see on https://github.com/reactjs/id.reactjs.org/pull/123/files/b293ec3a432e9bd035f16b07b719b873d108c2e0 is up. This is according to your feedback today (11.00 A.M, Oct 25).

I don't need to create new `pull request`, right?

Thank you

resir014 commented:
@dummyeuy You do not need to close the pull request, as you can continue making changes here until everything is good.

dummyeuy commented:
> @dummyeuy You do not need to close the pull request, as you can continue making changes here until everything is good.

Do I need to reopen this pull request?

resir014 commented:
Yes.

resir014 commented:
> Before I reopen this PR, do I have to close this JSX-in-depth PR or wait for that JSX-in-depth PR merged then reopen this one?

If you don't know how to remove the commit history to delete the Hooks FAQ file, then you can just close that PR and reopen. I'll only merge completed PRs.

dummyeuy commented:
> Yes.

@resir014 
Before I reopen this PR, do I have to close this [JSX-in-depth PR](https://github.com/reactjs/id.reactjs.org/pull/125) or wait for that JSX-in-depth PR merged then reopen this one?

dummyeuy commented:
> > Before I reopen this PR, do I have to close this JSX-in-depth PR or wait for that  JSX-in-depth PR PR merged then reopen this one?
> 
> If you don't know how to remove the commit history to delete the Hooks FAQ file, then you can just close that PR and reopen. I'll only merge completed PRs.

I've already closed Hooks API PR and I can't reopen the PR because of the [JSX-in-depth PR](https://github.com/reactjs/id.reactjs.org/pull/125) is still open -not merged yet

dummyeuy commented:
Excuse me @resir014   , do I have to make a change in this PR too?
*I've just reopened this PR

resir014 commented:
@dummyeuy I'm not sure I understand your question 😅 What do you mean by that?

dummyeuy commented:
> @dummyeuy I'm not sure I understand your question 😅 What do you mean by that?

@resir014 I mean "do I have to make some changes to this PR? because I'd closed this PR before - now I've just reopened this PR"

resir014 commented:
@dummyeuy 

> do I have to make some changes to this PR? because I'd closed this PR before - now I've just reopened this PR

Yes. However, on a1a815e04ee822d040f830bc1bbddc5aeb636ed5, I see that you've **deleted your work** on the Hooks FAQ page and replaced it with the merged translation from the other PR (JSX in-depth). I don't understand what you're doing here 😅

dummyeuy commented:
> @dummyeuy
> 
> > do I have to make some changes to this PR? because I'd closed this PR before - now I've just reopened this PR
> 
> Yes. However, on [a1a815e](https://github.com/reactjs/id.reactjs.org/commit/a1a815e04ee822d040f830bc1bbddc5aeb636ed5), I see that you've **deleted your work** on the Hooks FAQ page and replaced it with the merged translation from the other PR (JSX in-depth). I don't understand what you're doing here 😅

Hello @resir014 , I brought back my *deleted work* on **Hooks FAQ** before the JSX-in-depth PR.

You may check it out and review the translated version of Hooks FAQ on [my latest commit](https://github.com/dummyeuy/id.reactjs.org/commit/0a99663b39251dd4fc08f63f1067f71e0a93f1c6)

Thank you

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,148 +5,147 @@ permalink: docs/hooks-faq.html
+  * [Haruskah saya menggunakan Hooks, kelas atau gabungan keduanya?](#should-i-use-hooks-classes-or-a-mix-of-both)
+  * [Apakah *Hooks* mencakup seluruh kasus yang digunakan untuk kelas?](#do-hooks-cover-all-use-cases-for-classes)
+  * [Apakah *Hooks* menggantikan *render props* dan *higher-order components*?](#do-hooks-replace-render-props-and-higher-order-components)
+  * [Apa yang *Hooks* peruntukan untuk API populer seperti *Redux connect()* dan *React router*?](#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router)
```
```suggestion
  * [Apakah kegunaan *Hooks* untuk API populer seperti Redux connect() dan React router?](#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router)
```

- Library names/names of things do not need to be italicised.
- This translation slightly makes more sense.

dummyeuy replied:
>* [Apa kegunaan *Hooks* untuk API populer seperti Redux connect() dan React router?](#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router)

Suggested changes are applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,148 +5,147 @@ permalink: docs/hooks-faq.html
+Seringkali, *render props* dan *higher-order components* hanya (me)*render* sebuah turunan. Kami pikir *Hooks* adalah cara yang lebih sederhana untuk menyajikan kasus penggunaan ini. Masih ada tempat untuk kedua pola (contohnya, sebuah *virtual scroller component* bisa saja memiliki sebuah *prop* `renderItem`, atau sebuah *visual container component* bisa saja memiliki struktur *DOM*-nya sendiri) ...
 
-### What do Hooks mean for popular APIs like Redux `connect()` and React Router? {#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router}
+### Apa yang *Hooks* peruntukan untuk API populer seperti *Redux `connect()` dan *React router*? {#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router}
```
```suggestion
### Apakah kegunaan *Hooks* untuk API populer seperti Redux `connect()` dan React router? {#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router}
```

Idem.

dummyeuy replied:
> ### Apa kegunaan *Hooks* untuk API populer seperti Redux `connect()` dan React router? {#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router}

Suggested changes are applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,148 +5,147 @@ permalink: docs/hooks-faq.html
+  * [Apa saya harus menggunakan satu atau beberapa *state variable*?](#should-i-use-one-or-many-state-variables)
+  * [Bisakah saya menjalakan sebuah efek hanya pada pembaruan?](#can-i-run-an-effect-only-on-updates)
+  * [Bagimana cara untuk mendapatkan kembali *prop* atau *state* sebelumnya?](#how-to-get-the-previous-props-or-state)
+  * [Mengapa saya melihat *stale prop* atau *state* dalam fungsi yang saya buat?](#why-am-i-seeing-stale-props-or-state-inside-my-function)
```
```suggestion
  * [Mengapa saya melihat *prop* atau *state* yang basi dalam fungsi yang saya buat?](#why-am-i-seeing-stale-props-or-state-inside-my-function)
```

"stale" might be translatable to "basi" so let's go with that.

dummyeuy replied:
>   * [Mengapa saya melihat *prop* atau *state* yang basi dalam fungsi yang saya buat?](#why-am-i-seeing-stale-props-or-state-inside-my-function)

Suggested change's applied

dummyeuy replied:
> "stale" might be translatable to "basi" so let's go with that.

Understood. Thank you

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -376,21 +375,21 @@ function Counter() {
+Lihat juga [pola yang direkomendasikan untuk *derived state*](#how-do-i-implement-getderivedstatefromprops).
 
-### Why am I seeing stale props or state inside my function? {#why-am-i-seeing-stale-props-or-state-inside-my-function}
+### Mengapa saya melihat *stale prop* atau *state* dalam fungsi yang saya buat?  {#why-am-i-seeing-stale-props-or-state-inside-my-function}
```
```suggestion
### Mengapa saya melihat *prop* atau *state* yang basi dalam fungsi yang saya buat?  {#why-am-i-seeing-stale-props-or-state-inside-my-function}
```

dummyeuy replied:
> ### Mengapa saya melihat *prop* atau *state* yang basi dalam fungsi yang saya buat?  {#why-am-i-seeing-stale-props-or-state-inside-my-function}

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,148 +5,147 @@ permalink: docs/hooks-faq.html
+  * [Apakah *Hooks* mencakup seluruh kasus yang digunakan untuk kelas?](#do-hooks-cover-all-use-cases-for-classes)
+  * [Apakah *Hooks* menggantikan *render props* dan *higher-order components*?](#do-hooks-replace-render-props-and-higher-order-components)
+  * [Apa yang *Hooks* peruntukan untuk API populer seperti *Redux connect()* dan *React router*?](#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router)
+  * [Apakah *Hooks* dapat berjalan dengan *static typing*?](#do-hooks-work-with-static-typing)
```
```suggestion
  * [Apakah *Hooks* dapat bekerja dengan *static typing*?](#do-hooks-work-with-static-typing)
```

dummyeuy replied:
>   * [Apakah *Hooks* dapat bekerja dengan *static typing*?](#do-hooks-work-with-static-typing)

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,148 +5,147 @@ permalink: docs/hooks-faq.html
+*Library* seperti React Router akan mendukung *Hooks* di masa yang akan datang.
 
-### Do Hooks work with static typing? {#do-hooks-work-with-static-typing}
+### Apakah *Hooks* dapat berjalan dengan *static typing*?{#do-hooks-work-with-static-typing}
```
```suggestion
### Apakah *Hooks* dapat bekerja dengan *static typing*?{#do-hooks-work-with-static-typing}
```

dummyeuy replied:
> ### Apakah *Hooks* dapat bekerja dengan *static typing*?{#do-hooks-work-with-static-typing}

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,148 +5,147 @@ permalink: docs/hooks-faq.html
+### Apakah *Hooks* dapat berjalan dengan *static typing*?{#do-hooks-work-with-static-typing}
 
-Hooks were designed with static typing in mind. Because they're functions, they are easier to type correctly than patterns like higher-order components. The latest Flow and TypeScript React definitions include support for React Hooks.
+*Hooks* didesain dengan *static typing* pada dasarnya. Karena hal-hal tersebut adalah fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi *Flow* dan *TypeScript React* untuk React sudah mendukung untuk React Hooks di dalamnya.
```
```suggestion
*Hooks* didesain dengan *static typing* pada dasarnya. Karena *Hooks* adalah sekumpulan fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi Flow dan TypeScript untuk React sudah mendukung React Hooks.
```

- Ambiguous phrasing eliminated.
- Names of other languages (e.g. Flow and TypeScript) don't need to be italicised

dummyeuy replied:
>*Hooks* didesain dengan *static typing* pada dasarnya. Karena *Hooks* adalah sekumpulan fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi Flow dan TypeScript React untuk React sudah mendukung React Hooks.

Suggested changes are applied.

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,148 +5,147 @@ permalink: docs/hooks-faq.html
+### Apa yang *Hooks* peruntukan untuk API populer seperti *Redux `connect()` dan *React router*? {#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router}
 
-You can continue to use the exact same APIs as you always have; they'll continue to work.
+Anda dapat terus menggunakan *API* yang sama seperti yang selama ini dilakukan; *API* tersebut akan tetap bekerja.
```
```suggestion
Anda dapat terus menggunakan API yang sama seperti yang selama ini dilakukan; API tersebut akan tetap bekerja.
```

Please remove all italicisation of API. Abbreviations for foreign names don't need to be translated.

https://twitter.com/ivanlanin/status/898520073224167424

dummyeuy replied:
> Anda dapat terus menggunakan API yang sama seperti yang selama ini dilakukan; API tersebut akan tetap bekerja.

Suggested change's applied

dummyeuy replied:
> Please remove all italicisation of API. Abbreviations for foreign names don't need to be translated.
> 
> https://twitter.com/ivanlanin/status/898520073224167424

All italicization of `API` are changed.

dummyeuy commented:
Thank you @resir014  for your review and suggested changes.
* I have removed all _italicization of abbreviations of names_ (e.g. API, DOM)
* I have removed all spaces between all of the _spaces between the language name and line numbers_

Please kindly review my [latest commit](https://github.com/reactjs/id.reactjs.org/pull/123/commits/4863422610a2a600baa66cebaca14f3a59405fbc) :)
Thank you again

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,146 +5,145 @@ permalink: docs/hooks-faq.html
+### Haruskah saya menggunakan Hooks, kelas atau gabungan keduanya?{#should-i-use-hooks-classes-or-a-mix-of-both}
 
-When you're ready, we'd encourage you to start trying Hooks in new components you write. Make sure everyone on your team is on board with using them and familiar with this documentation. We don't recommend rewriting your existing classes to Hooks unless you planned to rewrite them anyway (e.g. to fix bugs).
+Ketika Anda siap, kami menganjurkan Anda untuk mulai mencoba *Hooks* dalam komponen-komponen baru yang Anda tulis. Pastikan setiap orang dalam tim Anda siap dan tidak asing dengan dokumentasi ini. Kami tidak merekomendasikan menulis ulang kelas-kelas Anda yang sudah ada ke *Hooks* kecuali Anda sudah berencana menulis ulang (seperti, memperbaiki *bug*).
```
```suggestion
Ketika Anda siap, kami menganjurkan Anda untuk mulai mencoba *Hooks* dalam komponen-komponen baru yang Anda tulis. Pastikan setiap orang dalam tim Anda siap dan tidak asing dengan dokumentasi ini. Kami tidak merekomendasikan menulis ulang kelas-kelas Anda yang sudah ada ke *Hooks* kecuali Anda sudah berencana menulis ulang (seperti memperbaiki *bug*).
```

No need the comma here.

dummyeuy replied:
>Ketika Anda siap, kami menganjurkan Anda untuk mulai mencoba *Hooks* dalam komponen-komponen baru yang Anda tulis. Pastikan setiap orang dalam tim Anda siap dan tidak asing dengan dokumentasi ini. Kami tidak merekomendasikan menulis ulang kelas-kelas Anda yang sudah ada ke *Hooks* kecuali Anda sudah berencana menulis ulang (seperti memperbaiki *bug*).

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,146 +5,145 @@ permalink: docs/hooks-faq.html
+### Apakah *Hooks* dapat bekerja dengan *static typing*?{#do-hooks-work-with-static-typing}
 
-Hooks were designed with static typing in mind. Because they're functions, they are easier to type correctly than patterns like higher-order components. The latest Flow and TypeScript React definitions include support for React Hooks.
+*Hooks* didesain dengan *static typing* pada dasarnya. Karena *Hooks* adalah sekumpulan fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi Flow dan TypeScript React untuk React sudah mendukung React Hooks.
```
```suggestion
*Hooks* didesain dengan *static typing* pada dasarnya. Karena *Hooks* adalah sekumpulan fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi Flow dan TypeScript untuk React sudah mendukung React Hooks.
```

Double mention of React in here, perhaps?

dummyeuy replied:
>*Hooks* didesain dengan *static typing* pada dasarnya. Karena *Hooks* adalah sekumpulan fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi Flow dan TypeScript untuk React sudah mendukung React Hooks.

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
````diff
@@ -5,146 +5,145 @@ permalink: docs/hooks-faq.html
 ```
-
-We'll test it using React DOM. To make sure that the behavior matches what happens in the browser, we'll wrap the code rendering and updating it into [`ReactTestUtils.act()`](/docs/test-utils.html#act) calls:
+Kami akan mengujinya dengan React DOM. Pastikan bahwa perilakunya dengan yang terjadi di peramban, kami akan mengemas proses *rendering* kode dan memperbaruinya ke dalam [`ReactTestUtils.act()`](/docs/test-utils.html#act) beberapa pemanggilan:
````
```suggestion

Kami akan mengujinya dengan React DOM. Pastikan bahwa perilakunya dengan yang terjadi di peramban, kami akan mengemas proses *rendering* kode dan memperbaruinya ke dalam [`ReactTestUtils.act()`](/docs/test-utils.html#act) beberapa pemanggilan:
```

Missing paragraph break here. This will cause merge conflicts when we sync new updates from the upstream reactjs.org docs.

dummyeuy replied:
Bring back missing paragraph break

resir014 commented on content/docs/hooks-faq.md:
````diff
@@ -256,11 +255,11 @@ If we just wanted to set an interval, we wouldn't need the ref (`id` could be lo
 ```
 
-Conceptually, you can think of refs as similar to instance variables in a class. Unless you're doing [lazy initialization](#how-to-create-expensive-objects-lazily), avoid setting refs during rendering -- this can lead to surprising behavior. Instead, typically you want to modify refs in event handlers and effects.
+Secara konsep, Anda bisa pikirkan *ref* itu serupa dengan *instance variables* dalam sebuah kelas. Kecuali, Anda menerapkan [lazy initialization](#how-to-create-expensive-objects-lazily), hindari pengaturan ref selama proses *render* -- hal ini akan mengakibatkan *behavior* yang tidak terduga. Sebagai gantinya, biasanya Anda ingin untuk memodifikasi *ref* dalam *event handler* dan efek.
````
```suggestion
Secara konsep, Anda bisa pikirkan *ref* itu serupa dengan variabel *instance* dalam sebuah kelas. Kecuali, Anda menerapkan [lazy initialization](#how-to-create-expensive-objects-lazily), hindari pengaturan ref selama proses *render* -- hal ini akan mengakibatkan *behavior* yang tidak terduga. Sebagai gantinya, biasanya Anda ingin untuk memodifikasi *ref* dalam *event handler* dan efek.
```

dummyeuy replied:
>Secara konsep, Anda bisa pikirkan *ref* itu serupa dengan variabel *instance* dalam sebuah kelas. Kecuali, Anda menerapkan [lazy initialization](#how-to-create-expensive-objects-lazily), hindari pengaturan ref selama proses *render* -- hal ini akan mengakibatkan *behavior* yang tidak terduga. Sebagai gantinya, biasanya Anda ingin untuk memodifikasi *ref* dalam *event handler* dan efek.

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -321,17 +320,17 @@ function useWindowPosition() {
+### Bisakah saya menjalakan sebuah efek hanya pada pembaruan? {#can-i-run-an-effect-only-on-updates}
 
-This is a rare use case. If you need it, you can [use a mutable ref](#is-there-something-like-instance-variables) to manually store a boolean value corresponding to whether you are on the first or a subsequent render, then check that flag in your effect. (If you find yourself doing this often, you could create a custom Hook for it.)
+Ini adalah kasus penggunaan yang langka. Jika butuh, Anda bisa [gunakan sebuah *ref* yang bisa diubah-ubah](#is-there-something-like-instance-variables) menjadi secara manual menyetorkan nilai *boolean* sesuai dengan apakah Anda berada pada *render* pertama atau selanjutnya, kemudia memeriksa penandanya (*flag*) pada efek yang Anda buat. (Jika sering melakukan hal ini, ada baiknya Anda membuat se ...
```
```suggestion
Ini adalah kasus penggunaan yang langka. Jika butuh, Anda dapat [menggunakan sebuah *ref* yang *mutable*](#is-there-something-like-instance-variables) untuk menyimpan secara manual nilai *boolean* sesuai dengan apakah Anda berada pada *render* pertama atau selanjutnya, kemudian memeriksa penandanya (*flag*) pada efek yang Anda buat. (Jika sering melakukan hal ini, ada baiknya Anda membuat sebuah *custom Hook*.)
```

This translation makes more sense, IMO.

dummyeuy replied:
>Ini adalah kasus penggunaan yang langka. Jika butuh, Anda dapat [menggunakan sebuah *ref* yang bisa *mutable*](#is-there-something-like-instance-variables)untuk menyimpan secara manual nilai *boolean* sesuai dengan apakah Anda berada pada *render* pertama atau selanjutnya, kemudian memeriksa penandanya (*flag*) pada efek yang Anda buat. (Jika sering melakukan hal ini, ada baiknya Anda membuat sebuah *custom Hook*.)

Suggested changes are applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -321,17 +320,17 @@ function useWindowPosition() {
+### Bagaimana cara untuk mendapatkan kembali *prop* atau *state* sebelumnya? {#how-to-get-the-previous-props-or-state}
 
-Currently, you can do it manually [with a ref](#is-there-something-like-instance-variables):
+Kini, Anda bisa melakukannya secara manual [dengan *ref*](#is-there-something-like-instance-variables):
```
```suggestion
Saat ini, Anda dapat melakukannya secara manual [dengan *ref*](#is-there-something-like-instance-variables):
```

dummyeuy replied:
>Saat ini, Anda dapat melakukannya secara manual [dengan *ref*](#is-there-something-like-instance-variables):

Suggested changes are applied

resir014 commented on content/docs/hooks-faq.md:
````diff
@@ -347,7 +346,7 @@ function Counter() {
 ```
 
-This might be a bit convoluted but you can extract it into a custom Hook:
+Ini bisa jadi agak berbelit tetapi Anda bisa mengekstraknya ke dalam sebuah *custom Hook*:
````
```suggestion
Cara ini mungkin agak rumit tetapi Anda bisa mengekstraknya ke dalam sebuah *custom Hook*:
```

dummyeuy replied:
>Cara ini mungkin agak rumit tetapi Anda bisa mengekstraknya ke dalam sebuah *custom Hook*:

Suggested changes are applied

resir014 commented on content/docs/hooks-faq.md:
````diff
@@ -365,7 +364,7 @@ function usePrevious(value) {
 ```
 
-Note how this would work for props, state, or any other calculated value.
+Perhatikan bagaimana hal ini dapat bekerja pada *props, state,* atau nilai apapun yang diperhitungkan.
````
```suggestion
Perhatikan bagaimana *Hook* ini dapat bekerja pada *props, state,* atau nilai apapun yang diperhitungkan.
```

dummyeuy replied:
>Perhatikan bagaimana *Hook* ini dapat bekerja pada *props, state,* atau nilai apapun yang diperhitungkan.

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
````diff
@@ -376,21 +375,21 @@ function Counter() {
 ```
 
-It's possible that in the future React will provide a `usePrevious` Hook out of the box since it's a relatively common use case.
+Di masa yang akan datang, React mungkin akan menyediakan sebuah `usePrevious` Hook* yang unik mengingat hal tersebut adalah kasus yang cukup sering digunakan.
````
```suggestion
Di masa yang akan datang, React mungkin akan menyediakan sebuah *Hook* `usePrevious` mengingat hal tersebut adalah kasus yang cukup sering digunakan.
```

dummyeuy replied:
>Di masa yang akan datang, React mungkin akan menyediakan sebuah *Hook* `usePrevious` mengingat hal tersebut adalah kasus yang cukup sering digunakan.

Suggested changes are applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -408,29 +407,29 @@ function Example() {
+>Catatan
 >
->We provide an [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) ESLint rule as a part of the [`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation) package. It warns when dependencies are specified incorrectly and suggests a fix.
+>Kami menyediakan sebuah aturan *ESLint* [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) sebagai bagian dari *package*[`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation). Itu akan memberi peringatan ketika *dependency* keliru dispesifikasikan dan akan memberikan saran perbaikan.
```
```suggestion
>Kami menyediakan sebuah aturan *ESLint* [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) sebagai bagian dari *package*[`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation). Aturan ini akan memberi peringatan ketika *dependency* keliru dispesifikasikan dan akan memberikan saran perbaikan.
```

Ambiguous "itu".

dummyeuy replied:
> >Kami menyediakan sebuah aturan *ESLint* [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) sebagai bagian dari *package*[`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation). Aturan ini akan memberi peringatan ketika *dependency* keliru dispesifikasikan dan akan memberikan saran perbaikan.

Suggested correction's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -439,13 +438,13 @@ function ScrollView({row}) {
+Hal ini bisa jadi terlihat aneh pada awalnya, tetapi sebuah pembaruan selama proses *render* adalah hal yang persis secara konsep sejak `getDerivedStateFromProps` dibuat.
 
-### Is there something like forceUpdate? {#is-there-something-like-forceupdate}
+### Apakah ada suatu hal seperti *forceUpdate*? {#is-there-something-like-forceupdate}
```
```suggestion
### Apakah ada fungsionalitas seperti *forceUpdate*? {#is-there-something-like-forceupdate}
```

dummyeuy replied:
>### Apakah ada fungsionalitas seperti *forceUpdate*? {#is-there-something-like-forceupdate}

Suggested change's applied.
Table of contents also updated

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,146 +5,145 @@ permalink: docs/hooks-faq.html
+  * [Bagimana cara untuk mendapatkan kembali *prop* atau *state* sebelumnya?](#how-to-get-the-previous-props-or-state)
+  * [Mengapa saya melihat *prop* atau *state* yang basi dalam fungsi yang saya buat?](#why-am-i-seeing-stale-props-or-state-inside-my-function)
+  * [Bagaimana cara mengimplementasikan *getDerivedStateFromProps*?](#how-do-i-implement-getderivedstatefromprops)
+  * [Apakah ada suatu hal seperti *forceUpdate*?](#is-there-something-like-forceupdate)
```
```suggestion
  * [Apakah ada fungsionalitas seperti *forceUpdate*?](#is-there-something-like-forceupdate)
```

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -455,15 +454,15 @@ Normally, you shouldn't mutate local state in React. However, as an escape hatch
+Saat Anda seharusnya tidak sering memerlukan hal ini, Anda bisa saja mengekspos beberapa metode perintah pada sebuah *parent component* dengan menggunakan [`useImperativeHandle`](/docs/hooks-reference.html#useimperativehandle) *Hook*.
 
-### How can I measure a DOM node? {#how-can-i-measure-a-dom-node}
+### Bagaimana cara mengukur *DOM node*? {#how-can-i-measure-a-dom-node}
```
```suggestion
### Bagaimana cara mengukur sebuah simpul DOM? {#how-can-i-measure-a-dom-node}
```

dummyeuy replied:
>### Bagaimana cara mengukur *node* DOM? {#how-can-i-measure-a-dom-node}

I keep the 'node' still as 'node', but remove the italicization of DOM.

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -5,146 +5,145 @@ permalink: docs/hooks-faq.html
+  * [Bagaimana cara mengimplementasikan *getDerivedStateFromProps*?](#how-do-i-implement-getderivedstatefromprops)
+  * [Apakah ada suatu hal seperti *forceUpdate*?](#is-there-something-like-forceupdate)
+  * [Bisakah saya membuat sebuah *ref* pada sebuah komponen fungsi?](#can-i-make-a-ref-to-a-function-component)
+  * [Bagaimana cara mengukur sebuah node *DOM*](#how-can-i-measure-a-dom-node)
```
```suggestion
  * [Bagaimana cara mengukur sebuah simpul DOM?](#how-can-i-measure-a-dom-node)
```

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -455,15 +454,15 @@ Normally, you shouldn't mutate local state in React. However, as an escape hatch
+### Bagaimana cara mengukur *DOM node*? {#how-can-i-measure-a-dom-node}
 
-In order to measure the position or size of a DOM node, you can use a [callback ref](/docs/refs-and-the-dom.html#callback-refs). React will call that callback whenever the ref gets attached to a different node. Here is a [small demo](https://codesandbox.io/s/l7m0v5x4v9):
+Untuk mengukur posisi atau ukuran dari sebuah *node* DOM, Anda bisa gunakan [*callback ref*](/docs/refs-and-the-dom.html#callback-refs). React akan memanggil *callback* itu kapanpun *ref* terikat pada sebuah *node* yang berbeda. Berikut ini [demo kecil](https://codesandbox.io/s/l7m0v5x4v9):
```
```suggestion
Untuk mengukur posisi atau ukuran dari sebuah simpul DOM, Anda bisa gunakan [*callback ref*](/docs/refs-and-the-dom.html#callback-refs). React akan memanggil *callback* itu kapanpun *ref* terikat pada sebuah *node* yang berbeda. Berikut ini [demo kecil](https://codesandbox.io/s/l7m0v5x4v9):
```

dummyeuy replied:
>* [Bagaimana cara mengukur sebuah node *DOM*](#how-can-i-measure-a-dom-node)

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -565,59 +564,59 @@ useEffect(() => {
+>Catatan
 >
->We provide the [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) ESLint rule as a part of the [`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation) package. It helps you find components that don't handle updates consistently.
+>Kami menyediakan aturan *ESLint* [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) sebagai bagian dari *package* [`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation). Itu akan membantu Anda menemukan komponen yang tidak konsisten menangani pembaruan.
```
```suggestion
>Kami menyediakan aturan *ESLint* [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) sebagai bagian dari *package* [`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation). Aturan ini akan membantu Anda menemukan komponen yang tidak konsisten menangani pembaruan.
```

Ambiguous "itu".

dummyeuy replied:
>>Kami menyediakan aturan *ESLint* [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) sebagai bagian dari *package* [`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation). Aturan ini akan membantu Anda menemukan komponen yang tidak konsisten menangani pembaruan.

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -633,135 +632,135 @@ This also allows you to handle out-of-order responses with a local variable insi
+### Bagaimana cara *memoize* perhitungan? {#how-to-memoize-calculations}
 
-The [`useMemo`](/docs/hooks-reference.html#usememo) Hook lets you cache calculations between multiple renders by "remembering" the previous computation:
+*Hook* [`useMemo`](/docs/hooks-reference.html#usememo) mengizinkan Anda untuk menyimpan (*cache*) perhitungan-perhitungan di antara berbagai *render* dengan cara *"mengingat"* komputasi sebelumnya:
```
```suggestion
*Hook* [`useMemo`](/docs/hooks-reference.html#usememo) memungkinkan Anda untuk menyimpan (*cache*) perhitungan-perhitungan di antara berbagai *render* dengan cara *"mengingat"* komputasi sebelumnya:
```

dummyeuy replied:
>*Hook* [`useMemo`](/docs/hooks-reference.html#usememo) memungkinkan Anda untuk menyimpan (*cache*) perhitungan-perhitungan di antara berbagai *render* dengan cara *"mengingat"* komputasi sebelumnya:

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -778,102 +777,102 @@ function Parent({ a, b }) {
+* Pada akhirnya, *Hook* [`useReducer`](/docs/hooks-reference.html#usereducer) mengurangi kebutuhan untuk mengoper *callback* secara mendalam, seperti yang dijelaskan di bawah ini.
 
-### How to avoid passing callbacks down? {#how-to-avoid-passing-callbacks-down}
+### Bagaimana cara menghindari pengoperan *callback* jadi *down*? {#how-to-avoid-passing-callbacks-down}
```
```suggestion
### Bagaimana cara menghindari pengoperan *callback* ke bawah? {#how-to-avoid-passing-callbacks-down}
```

This refers to passing callbacks down the node. Not actually putting the callback *down*.

dummyeuy replied:
>### Bagaimana cara menghindari pengoperan *callback* ke bawah? {#how-to-avoid-passing-callbacks-down}

Suggested correction's applied.
Table of contents also updated

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -778,102 +777,102 @@ function Parent({ a, b }) {
+### Bagaimana cara menghindari pengoperan *callback* jadi *down*? {#how-to-avoid-passing-callbacks-down}
 
-We've found that most people don't enjoy manually passing callbacks through every level of a component tree. Even though it is more explicit, it can feel like a lot of "plumbing".
+Kami menemukan bahwa kebanyakan orang tidak menikmati pengoperan *callback* secara manual di sepanjang tiap tingkatan sebuah pohon komponen. Walaupun hal tersebut lebih gamblang, pengoperan *callback* secara manual bisa terasa seperti pekerjaan yang menguras tenaga.
```
```suggestion
Kami menemukan bahwa kebanyakan orang tidak menikmati pengoperan *callback* secara manual di sepanjang tiap tingkatan sebuah diagram komponen. Walaupun hal tersebut lebih gamblang, pengoperan *callback* secara manual bisa terasa seperti pekerjaan yang menguras tenaga.
```

dummyeuy replied:
>Kami menemukan bahwa kebanyakan orang tidak menikmati pengoperan *callback* secara manual di sepanjang tiap tingkatan sebuah diagram komponen. Walaupun hal tersebut lebih gamblang, pengoperan *callback* secara manual bisa terasa seperti pekerjaan yang menguras tenaga.

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -778,102 +777,102 @@ function Parent({ a, b }) {
+Kami menemukan bahwa kebanyakan orang tidak menikmati pengoperan *callback* secara manual di sepanjang tiap tingkatan sebuah pohon komponen. Walaupun hal tersebut lebih gamblang, pengoperan *callback* secara manual bisa terasa seperti pekerjaan yang menguras tenaga.
 
-In large component trees, an alternative we recommend is to pass down a `dispatch` function from [`useReducer`](/docs/hooks-reference.html#usereducer) via context:
+Dalam pohon komponen yang luas, sebuah alternatif yang kami rekomendasikan adalah untuk mengoper sebuah fungsi `dispatch` dari [`useReducer`](/docs/hooks-reference.html#usereducer) melalui *context*:
```
```suggestion
Dalam diagram komponen yang luas, sebuah alternatif yang kami rekomendasikan adalah untuk mengoper sebuah fungsi `dispatch` dari [`useReducer`](/docs/hooks-reference.html#usereducer) melalui *context*:
```

`pohon` -> `diagram`.

dummyeuy replied:
>Dalam diagram komponen yang luas, sebuah alternatif yang kami rekomendasikan adalah untuk mengoper sebuah fungsi `dispatch` dari [`useReducer`](/docs/hooks-reference.html#usereducer) melalui *context*:

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
````diff
@@ -884,11 +883,11 @@ function TodosApp() {
 ```
 
-Any child in the tree inside `TodosApp` can use the `dispatch` function to pass actions up to `TodosApp`:
+Turunan apapun dalam pohon yang ada di dalam `TodosApp` bisa menggunakan fungsi `dispatch` untuk mengoper tindakan kepada `TodosApp`:
````
```suggestion
Turunan apapun dalam diagram yang ada di dalam `TodosApp` bisa menggunakan fungsi `dispatch` untuk mengoper tindakan kepada `TodosApp`:
```

dummyeuy replied:
>Turunan apapun dalam diagram yang ada di dalam `TodosApp` bisa menggunakan fungsi `dispatch` untuk mengoper tindakan kepada `TodosApp`:

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -901,33 +900,33 @@ function DeepChild(props) {
+>Catatan
 >
->We recommend to [pass `dispatch` down in context](#how-to-avoid-passing-callbacks-down) rather than individual callbacks in props. The approach below is only mentioned here for completeness and as an escape hatch.
+>Kami rekomendasikan untuk [mengoper `dispatch` dalam *context*](#how-to-avoid-passing-callbacks-down) daripada satu-persatu *callback* dalam *prop*. Pendekatan di bawah hanya disebutkan di sini sebagai pelengkap dan juga cara darurat.
```
```suggestion
>Kami rekomendasikan untuk [mengoper `dispatch` dalam *context*](#how-to-avoid-passing-callbacks-down) daripada *callback* dalam *prop* satu-persatu. Pendekatan di bawah hanya disebutkan di sini sebagai pelengkap dan juga cara darurat.
```

dummyeuy replied:
>>Kami rekomendasikan untuk [mengoper `dispatch` dalam *context*](#how-to-avoid-passing-callbacks-down) daripada *callback* dalam *prop* satu-persatu. Pendekatan di bawah hanya disebutkan di sini sebagai pelengkap dan juga cara darurat.

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -938,12 +937,12 @@ function Form() {
 function Form() {
   const [text, updateText] = useState('');
-  // Will be memoized even if `text` changes:
+  // Akan di memoize bahkan jika `text` berubah:
```
```suggestion
  // Akan di-memoize bahkan jika `text` berubah:
```

dummyeuy replied:
> // Akan di-memoize bahkan jika `text` berubah:

Suggested hyphen addition's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -972,27 +971,27 @@ function useEventCallback(fn, dependencies) {
+### Bagaimana React menghubungkan panggilan *Hook* dengan komponen? {#how-does-react-associate-hook-calls-with-components}
 
-React keeps track of the currently rendering component. Thanks to the [Rules of Hooks](/docs/hooks-rules.html), we know that Hooks are only called from React components (or custom Hooks -- which are also only called from React components).
+React merekam jejak komponen yang baru saja mulai di-*render*. Terima kasih pada [aturan *Hooks*](/docs/hooks-rules.html), kita tahu bahwa *Hooks* hanya dipanggil dari komponen React (atau *custom Hooks* -- yang mana juga dipanggil dari komponen React).
```
```suggestion
React merekam jejak komponen yang baru saja mulai di-*render*. Berkat [aturan *Hooks*](/docs/hooks-rules.html), kita tahu bahwa *Hooks* hanya dipanggil dari komponen React (atau *custom Hooks* -- yang mana juga dipanggil dari komponen React).
```

dummyeuy replied:
>React merekam jejak komponen yang baru saja mulai di-*render*. Berkat [aturan *Hooks*](/docs/hooks-rules.html), kita tahu bahwa *Hooks* hanya dipanggil dari komponen React (atau *custom Hooks* -- yang mana juga dipanggil dari komponen React).

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -972,27 +971,27 @@ function useEventCallback(fn, dependencies) {
-* [Subscriptions](http://reactivex.io/rxjs/class/es6/Subscription.js~Subscription.html) in Rx.
-* [Algebraic effects](https://github.com/ocamllabs/ocaml-effects-tutorial#2-effectful-computations-in-a-pure-setting) in Multicore OCaml.
+* Eksperimen lama kami dengan API fungsional dalam repositori[*react-future*](https://github.com/reactjs/react-future/tree/master/07%20-%20Returning%20State).
+* Eksperimen-eksperimen komunitas React dengan me-*render prop API*, termasuk [Reactions Component](https://github.com/reactions/component) dari [Ryan Florence](https://github.com/ryanflorence).
```
```suggestion
* Eksperimen-eksperimen komunitas React dengan me-*render* API *props*, termasuk [Reactions Component](https://github.com/reactions/component) dari [Ryan Florence](https://github.com/ryanflorence).
```

dummyeuy replied:
>* Eksperimen-eksperimen komunitas React dengan me-*render* API *props*, termasuk [Reactions Component](https://github.com/reactions/component) dari [Ryan Florence](https://github.com/ryanflorence).

Suggested change's applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -972,27 +971,27 @@ function useEventCallback(fn, dependencies) {
-* [Reducer components](https://reasonml.github.io/reason-react/docs/en/state-actions-reducer.html) in ReasonReact.
-* [Subscriptions](http://reactivex.io/rxjs/class/es6/Subscription.js~Subscription.html) in Rx.
-* [Algebraic effects](https://github.com/ocamllabs/ocaml-effects-tutorial#2-effectful-computations-in-a-pure-setting) in Multicore OCaml.
+* Eksperimen lama kami dengan API fungsional dalam repositori[*react-future*](https://github.com/reactjs/react-future/tree/master/07%20-%20Returning%20State).
```
```suggestion
* Eksperimen lama kami dengan API fungsional dalam repositori [*react-future*](https://github.com/reactjs/react-future/tree/master/07%20-%20Returning%20State).
```

Missing space.

dummyeuy replied:
>* Eksperimen lama kami dengan API fungsional dalam repositori [*react-future*](https://github.com/reactjs/react-future/tree/master/07%20-%20Returning%20State).

Added missing space

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -972,27 +971,27 @@ function useEventCallback(fn, dependencies) {
+* *State variable* dan *state cell* dalam [DisplayScript](http://displayscript.org/introduction.html).
+* [Reducer components](https://reasonml.github.io/reason-react/docs/en/state-actions-reducer.html) dalam *ReasonReact*.
+* [Subscriptions](http://reactivex.io/rxjs/class/es6/Subscription.js~Subscription.html) dalam Rx.
+* [Algebraic effects](https://github.com/ocamllabs/ocaml-effects-tutorial#2-effectful-computations-in-a-pure-setting) dalam *Multicore OCaml*.
```
```suggestion
* [Algebraic effects](https://github.com/ocamllabs/ocaml-effects-tutorial#2-effectful-computations-in-a-pure-setting) dalam *Multicore* OCaml.
```

OCaml is a language name. No need to be italicised.

dummyeuy replied:
>* [Algebraic effects](https://github.com/ocamllabs/ocaml-effects-tutorial#2-effectful-computations-in-a-pure-setting) dalam *Multicore* OCaml.

Suggested correction is applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -972,27 +971,27 @@ function useEventCallback(fn, dependencies) {
+* Eksperimen-eksperimen komunitas React dengan me-*render prop API*, termasuk [Reactions Component](https://github.com/reactions/component) dari [Ryan Florence](https://github.com/ryanflorence).
+* Pengajuan [kata kunci `adopt`](https://gist.github.com/trueadm/17beb64288e30192f3aa29cad0218067) dari [Dominic Gannaway](https://github.com/trueadm) sebagai *sugar syntax* untuk me-*render* *prop*.
+* *State variable* dan *state cell* dalam [DisplayScript](http://displayscript.org/introduction.html).
+* [Reducer components](https://reasonml.github.io/reason-react/docs/en/state-actions-reducer.html) dalam *ReasonReact*.
```
```suggestion
* [Reducer components](https://reasonml.github.io/reason-react/docs/en/state-actions-reducer.html) dalam ReasonReact.
```

ReasonReact is a name of a library. No need to be italicised.

dummyeuy replied:
>* [Reducer components](https://reasonml.github.io/reason-react/docs/en/state-actions-reducer.html) dalam ReasonReact.

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -972,27 +971,27 @@ function useEventCallback(fn, dependencies) {
+* [Algebraic effects](https://github.com/ocamllabs/ocaml-effects-tutorial#2-effectful-computations-in-a-pure-setting) dalam *Multicore OCaml*.
 
-[Sebastian Markbåge](https://github.com/sebmarkbage) came up with the original design for Hooks, later refined by [Andrew Clark](https://github.com/acdlite), [Sophie Alpert](https://github.com/sophiebits), [Dominic Gannaway](https://github.com/trueadm), and other members of the React team.
+[Sebastian Markbåge](https://github.com/sebmarkbage) muncul dengan desain orisinil untuk *Hooks*, yang nantinya akan disempurnakan oleh [Andrew Clark](https://github.com/acdlite), [Sophie Alpert](https://github.com/sophiebits), [Dominic Gannaway](https://github.com/trueadm), serta anggota-anggota lain tim React.
```
```suggestion
[Sebastian Markbåge](https://github.com/sebmarkbage) membuat desain awal dari *Hooks*, yang kemudian disempurnakan oleh [Andrew Clark](https://github.com/acdlite), [Sophie Alpert](https://github.com/sophiebits), [Dominic Gannaway](https://github.com/trueadm), serta anggota-anggota lain tim React.
```

We're talking in the past tense. It *has been* refined by Andrew Clark.

dummyeuy replied:
>[Sebastian Markbåge](https://github.com/sebmarkbage) membuat desain awal dari *Hooks*, yang kemudian akan oleh [Andrew Clark](https://github.com/acdlite), [Sophie Alpert](https://github.com/sophiebits), [Dominic Gannaway](https://github.com/trueadm), serta anggota-anggota lain tim React.

Suggested corrections are applied

resir014 commented on content/docs/hooks-faq.md:
```diff
@@ -972,27 +971,27 @@ function useEventCallback(fn, dependencies) {
+Terdapat daftar internal dari *"memory cells"* yang berhubungkan dengan tiap komponen. Itu semua hanyalah objek-objek JavaScript dimana kita bisa menaruh data. Ketika Anda memanggil sebuah *Hook* seperti `useState()`, ia membaca *cell* terkini (atau menginisialisasikannya selama *render* pertama), dan memindahkan *pointer* ke *cell* selanjutnya. Inilah bagaimana masing-masing panggilan `useState( ...
 
-### What is the prior art for Hooks? {#what-is-the-prior-art-for-hooks}
+### Bagaimana *seni awal*-nya *Hooks*? {#what-is-the-prior-art-for-hooks}
```
```suggestion
### Adakah hasil kerja sebelumnya yang menginspirasi *Hooks*? {#what-is-the-prior-art-for-hooks}
```

dummyeuy replied:
>### Adakah hasil kerja sebelumnya yang menginspirasi *Hooks*? {#what-is-the-prior-art-for-hooks}

Suggested translation is applied.
Table of contents also updated

dummyeuy commented:
All suggested corrections are applied, and the headings & table of contents are also updated.
Merge conflict's resolved.
